src/contract.rs

```diff
@@ -65,18 +65,29 @@ pub fn query<S: Storage, A: Api, Q: Querier>(
         QueryMsg::Holder { address } => to_binary(&query_holder(&deps, address)?),
         QueryMsg::Holders { start_after, limit } => {
             to_binary(&query_holders(&deps, start_after, limit)?)
-        }
+        },
+        QueryMsg::TransferFrom { .. } => to_binary(&query_transfer_from(deps)?),
     }
 }
 
+fn query_transfer_from<S: Storage, A: Api, Q: Querier>(
```

Comment:
`TransferFrom` is an execution function and not related to querying, Also why you need TransferFrom in this contract anyways. I mean the transfer operations are in staking cw20 contract?

Reply:
We need allowance in order to move tokens from an user to our staking contract or you have another solution ? 
We still need a struct to encode the msg

Reply:
https://github.com/LoTerra/loterra-staking-contract/pull/1#discussion_r618604396